pkg/resolvers/k8s/streamer.go

```diff
@@ -50,16 +51,20 @@ func startNewStreamer(target targetEntry, epClient endpointClient) (*streamer, e
 		cancel:   cancel,
 	}
 	firstUpdateCh := make(chan struct{})
+
+	// Schedule go routine that gets forward and translates stream messages as endpoint changes.
 	go func() {
 		defer cancel()
 
 		if err := proxy(ctx, json.NewDecoder(stream), s.changeCh, firstUpdateCh); err != nil {
 			s.errCh <- err
 		}
 	}()
+
+	// Try to return valid streamer only when we have the first update. Otherwise we might have empty resolution for some initial time.
 	select {
 	case <-firstUpdateCh:
-	case <-time.After(100 * time.Millisecond):
+	case <-time.After(300 * time.Millisecond):
```

Comment:
If this change is necessary, this should probably be a flag.

Reply:
yup